src/GraphQL/ResolverRegistryInterface.php

```diff
@@ -50,13 +51,24 @@ public function addFieldResolver($type, $field, ResolverInterface $resolver);
   /**
    * Return the field resolver for a given type and field name.
    *
+   * Most code will probably want to use `getFieldResolverWithInheritance`
+   * instead.
+   *
    * @param string $type
    * @param string $field
    *
    * @return callable|null
    */
   public function getFieldResolver($type, $field);
 
+  /**
+   * Return all field resolvers in the registry.
+   *
+   * @return callable[]
+   *   A nested list of callables, keyed by type and field name.
+   */
+  public function getAllFieldResolvers() : array;
```

Comment:
usually changing interfaces is a no-go in a stable branch, we would have to start 5.x for that.

I'm wondering how many users of the module implement this interface (probably not many). I still think we should not risk that and not change the interface. I think we can get away with just adding the new methods to the implementation? And then leave a todo comment on the interface to add them for 5.x.

Reply:
I thought about it but makes the calling code a bit more difficult, since it can't just rely on the type but has to actively check and throw an error. That makes that slightly more difficult.

I'll make the change.